examples/type_conversion.rs

```diff
@@ -0,0 +1,74 @@
+use ndarray::prelude::*;
+
+fn main() {
+    // To convert from one datatype in an array to another, we use the
+    // map-by-value functionality on the array. We specify the conversion as a
+    // closure to be applied to each element independently.
+    // This allows for more control and flexibility in converting types.
+    // Examples of common conversions are given below. Some conversions that
+    // might be surprising are shown as well, although the behavior in those
+    // cases are part of Rust's built-in conversions and are not specific to
+    // ndarray. For a brush-up on casting between numeric types in Rust, check
+    // out:
+    // https://doc.rust-lang.org/rust-by-example/types/cast.html
+
+    // Simple upcasting
+    let a_u8: Array<u8, _> = array![[1, 2, 3], [4, 5, 6]];
+    let a_u32 = a_u8.mapv(|element| element as u32);
+    assert_eq!(a_u32, array![[1u32, 2u32, 3u32], [4u32, 5u32, 6u32]]);
+
+    // Float to int casting - saturating
+    // The `as` keyword performs a *saturating cast* since Rust 1.45, which is
+    // slightly less performant than an unchecked conversion.
+    let a_f32: Array<f32, _> = array![
+        256.0,         // saturated to 255
+        255.7,         // saturated to 255
+        255.1,         // saturated to 255
+        254.7,         // rounded down to 254 by cutting the decimal part
+        254.1,         // rounded down to 254 by cutting the decimal part
+        -1.0,          // saturated to 0 on the lower end
+        f32::INFINITY, // saturated to 255
+        f32::NAN,      // converted to zero
+    ];
+    let a_u8 = a_f32.mapv(|element| element as u8);
+    assert_eq!(a_u8, array![255, 255, 255, 254, 254, 0, 255, 0]);
+
+    // Float to int casting - unchecked
+    // For a more performant unchecked conversion, we can use
+    // `f32::to_int_unchecked()`. Note that this requires an `unsafe` block.
+    let a_f32: Array<f32, _> = array![
+        257.0,         // 1, last bit is one
+        256.0,         // 0, last 8 bits are zero
+        254.0,         // 254, in bounds
+        -1.1,          // 255, reinterpreted 2-complement after cutting decimals
+        -2.3,          // 254, reinterpreted 2-complement after cutting decimals
+        -3.7,          // 253, reinterpreted 2-complement after cutting decimals
+        f32::INFINITY, // 0, based on representation
+        f32::NAN,      // 0, based on representation
+    ];
+    unsafe {
+        let a_u8: Array<u8, _> = a_f32.mapv(|element| element.to_int_unchecked::<u8>());
```

Comment:
This code is UB :boom:  so the example is meaningless :slightly_frowning_face:   Unfortunately.

Reply:
Alright thank you a lot for pointing that out - I will remove the example tomorrow :+1: